### PR TITLE
Add "Don't give the file editing message" flag in shortmess option.

### PR DESCRIPTION
**Pull request by Shougo**
It adds F flag in shortmess option.
It is "Don't give the file editing message" flag.
Because, it is useless if you use statusline feature.

### CODE REVIEW

File: src/buffer.c [hunk 1]
```diff
@@ -139,14 +139,20 @@ open_buffer(
 #endif
        )
     {
+        int old_msg_silent= msg_silent;
```
Comment (cvwillegen) [inline src/buffer.c]: Add a space there before the =

Comment (Shougo) [inline src/buffer.c]: Thank you.  Fixed.

**Comment by brammool** [comment]
Shougo wrote:

> It adds F flag in shortmess option.
> It is "Don't give the file editing message" flag.
> Because, it is useless if you use statusline feature.
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/vim/vim/pull/686
> 
> -- Commit Summary --
> - Add "Don't give the file editing message" flag in shortmess option.
> 
> -- File Changes --
> 
> ```
> M runtime/doc/options.txt (1)
> M src/buffer.c (6)
> M src/ex_cmds.c (6)
> M src/option.h (3)
> ```

Thanks, I'll include it.

## 

Microsoft is to software what McDonalds is to gourmet cooking

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by vim-ml** [comment]
Hi Bram,

2016-3-15(Tue) 22:31:01 UTC+9 Bram Moolenaar:

> Shougo wrote:
> 
> > It adds F flag in shortmess option.
> > 
> > It is "Don't give the file editing message" flag.
> > 
> > Because, it is useless if you use statusline feature.
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> >   https://github.com/vim/vim/pull/686
> > 
> > -- Commit Summary --
> > - Add "Don't give the file editing message" flag in shortmess option.
> > 
> > -- File Changes --
> > 
> > ```
> > M runtime/doc/options.txt (1)
> > 
> > M src/buffer.c (6)
> > 
> > M src/ex_cmds.c (6)
> > 
> > M src/option.h (3)
> > ```
> 
> Thanks, I'll include it.

---

src/buffer.c
@@ -139,14 +139,20 @@ open_buffer(
  #endif
         )
      {
- ```
     int old_msg_silent = msg_silent;
  ```
  
  +
  #ifdef FEAT_NETBEANS_INTG
    int oldFire = netbeansFireChanges;
  
    netbeansFireChanges = 0;
  #endif
-        if (shortmess(SHM_FILEINFO))
-            msg_silent = 1;
  retval = readfile(curbuf->b_ffname, curbuf->b_fname,
        (linenr_T)0, (linenr_T)0, (linenr_T)MAXLNUM, eap,
        flags | READ_NEW);
-        if (shortmess(SHM_FILEINFO))
-            msg_silent = old_msg_silent;
  #ifdef FEAT_NETBEANS_INTG
    netbeansFireChanges = oldFire;
  #endif
  --------

Once because the `shortmess(SHM_FILEINFO)` of the results are likely to change in autocmd in the readfile(), It should set to the variable.

int has_shm_fileinfo = shortmess(SHM_FILEINFO);

if (has_shm_fileinfo)
    msg_silent = 1;
retval = readfile(.......);
if (has_shm_fileinfo)
    msg_silent = old_msg_silent;

## 

Best regards,
Hirohito Higashi (a.k.a. h_east)

File: src/buffer.c (hunk 1 above)
Comment (brammool) [inline src/buffer.c]: cvwillegen wrote:

> > @@ -139,14 +139,20 @@ open_buffer(
> >  #endif
> >         )
> >      {
> > -        int old_msg_silent= msg_silent;
> 
> Add a space there before the =

Already done.

## 

hundred-and-one symptoms of being an internet addict:
44. Your friends no longer send you e-mail...they just log on to your IRC
    channel.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by brammool** [comment]
Hirohito Higashi wrote:

> ---
> 
> src/buffer.c
> @@ -139,14 +139,20 @@ open_buffer(
>   #endif
>          )
>       {
> - ```
>      int old_msg_silent = msg_silent;
>   ```
>   
>   +
>   #ifdef FEAT_NETBEANS_INTG
>       int oldFire = netbeansFireChanges;
>   
>   ```
>   netbeansFireChanges = 0;
>   ```
>   
>   #endif
> -        if (shortmess(SHM_FILEINFO))
> -            msg_silent = 1;
>     retval = readfile(curbuf->b_ffname, curbuf->b_fname,
>           (linenr_T)0, (linenr_T)0, (linenr_T)MAXLNUM, eap,
>           flags | READ_NEW);
> -        if (shortmess(SHM_FILEINFO))
> -            msg_silent = old_msg_silent;
>   #ifdef FEAT_NETBEANS_INTG
>       netbeansFireChanges = oldFire;
>   #endif
>   --------
> 
> Once because the `shortmess(SHM_FILEINFO)` of the results are likely to change in autocmd in the readfile(), It should set to the variable.
> 
> int has_shm_fileinfo = shortmess(SHM_FILEINFO);
> 
> if (has_shm_fileinfo)
>     msg_silent = 1;
> retval = readfile(.......);
> if (has_shm_fileinfo)
>     msg_silent = old_msg_silent;

I had solved that by restoring old_msg_silent always.  I don't think
msg_silent should change in readfile().

## 

hundred-and-one symptoms of being an internet addict:
45. You buy a Captain Kirk chair with a built-in keyboard and mouse.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by Shougo** [comment]
Thank you!

**Comment by vim-ml** [comment]
Hi Bram.

2016-3-16(Wed) 0:54:25 UTC+9 Bram:

> Hirohito Higashi wrote:
> 
> > ---
> > 
> > src/buffer.c
> > 
> > @@ -139,14 +139,20 @@ open_buffer(
> > 
> >   #endif
> > 
> > ```
> >      )
> > 
> >   {
> > ```
> > - ```
> >      int old_msg_silent = msg_silent;
> >   ```
> >   
> >   +
> >   
> >   #ifdef FEAT_NETBEANS_INTG
> >   
> >     int oldFire = netbeansFireChanges;
> >   
> >     netbeansFireChanges = 0;
> >   
> >   #endif
> > - ```
> >      if (shortmess(SHM_FILEINFO))
> >   ```
> > - ```
> >          msg_silent = 1;
> >   ```
> >   
> >   retval = readfile(curbuf->b_ffname, curbuf->b_fname,
> >   
> >   ```
> >     (linenr_T)0, (linenr_T)0, (linenr_T)MAXLNUM, eap,
> >   
> >     flags | READ_NEW);
> >   ```
> > - ```
> >      if (shortmess(SHM_FILEINFO))
> >   ```
> > - ```
> >          msg_silent = old_msg_silent;
> >   ```
> >   
> >   #ifdef FEAT_NETBEANS_INTG
> >   
> >     netbeansFireChanges = oldFire;
> >   
> >   #endif
> > 
> > ---
> > 
> > Once because the `shortmess(SHM_FILEINFO)` of the results are likely to change in autocmd in the readfile(), It should set to the variable.
> > 
> > int has_shm_fileinfo = shortmess(SHM_FILEINFO);
> > 
> > if (has_shm_fileinfo)
> > 
> > ```
> > msg_silent = 1;
> > ```
> > 
> > retval = readfile(.......);
> > 
> > if (has_shm_fileinfo)
> > 
> > ```
> > msg_silent = old_msg_silent;
> > ```
> 
> I had solved that by restoring old_msg_silent always.  I don't think
> 
> msg_silent should change in readfile().

Yes. Your fix was solved problem.

## Thanks.

Best regards,
Hirohito Higashi (a.k.a. h_east)

**Comment by Asheq** [comment]
I created a related issue: #3221

**Pull request closed without merging**